### PR TITLE
Synthetic pss support

### PR DESCRIPTION
**Pull request by williamcroberts**

### CODE REVIEW

File: src/lib/mech.c [hunk 1]
```diff
@@ -708,6 +709,92 @@ CK_RV rsa_pkcs_synthesizer(CK_MECHANISM_PTR mech, attr_list *attrs,
...
+
+    if (modulus_attr->ulValueLen > *outlen) {
+        LOGE("Output buffer is too small, got: %lu, required at least %lu",
+                *outlen, modulus_attr->ulValueLen);
+        return CKR_GENERAL_ERROR;
+    }
+
+    RSA *rsa = RSA_new();
+    if (!rsa) {
+        LOGE("oom");
```
Comment (williamcroberts) [inline src/lib/mech.c]: Can lean rsa

File: src/lib/tpm.c [hunk 2]
```diff
@@ -3162,26 +3162,60 @@ CK_RV tpm2_getmechanisms(tpm_ctx *ctx, CK_MECHANISM_TYPE *mechanism_list, CK_ULO
...
+    if (rv != CKR_OK) {
+        LOGE("Could not get fixed properties from TPM");
+        Esys_Free(capabilityData);
+        return rv;
+    }
+
+    TPMA_MODES modes = 0;
+    TPML_TAGGED_TPM_PROPERTY *plist = &fixed_props->data.tpmProperties;
+    UINT32 i;
+    for (i = 0; i < plist->count; ++i) {
```
Comment (williamcroberts) [inline src/lib/tpm.c]: i++

File: test/integration/pkcs-get-mechanism.int.c [hunk 3]
```diff
@@ -142,7 +142,6 @@ void test_get_mechanism_info_good(void **state) {
         { CKM_SHA1_RSA_PKCS,         CKF_HW | CKF_SIGN    | CKF_VERIFY                          },
         { CKM_SHA256_RSA_PKCS,       CKF_HW | CKF_SIGN    | CKF_VERIFY                          },
         { CKM_SHA384_RSA_PKCS,       CKF_HW | CKF_SIGN    | CKF_VERIFY                          },
-        { CKM_SHA512_RSA_PKCS,       CKF_HW | CKF_SIGN    | CKF_VERIFY                          },
```
Comment (williamcroberts) [inline test/integration/pkcs-get-mechanism.int.c]: Really? Split this to a separate patch and explain it.

File: src/lib/tpm.c [hunk 4]
```diff
@@ -3162,26 +3162,60 @@ CK_RV tpm2_getmechanisms(tpm_ctx *ctx, CK_MECHANISM_TYPE *mechanism_list, CK_ULO
         LOGE("Retrieving supported algorithms from TPM failed");
         return rv;
     }
+
     TPMU_CAPABILITIES *algs= &capabilityData->data;
```
Comment (williamcroberts) [inline src/lib/tpm.c]: Superfluous, remove

File: src/lib/mech.c (hunk 1 above)
Comment (williamcroberts) [reply src/lib/mech.c]: We can also drop the exponent, as we don't need it. Internally the code just needs n.

Comment (williamcroberts) [reply src/lib/mech.c]: Nope can't drop e, set0key will fail.

**Comment by pdxjohnny** [comment]
This solves the issue for #403

**Comment by dnoliver** [comment]
Can I test this with the test radius setup in Fedora 31 and expect it to work https://github.com/tpm2-software/tpm2-pkcs11/pull/403#issuecomment-584224098 ? :D

**Comment by williamcroberts** [comment]
> Can I test this with the test radius setup in Fedora 31 and expect it to work [#403 (comment)](https://github.com/tpm2-software/tpm2-pkcs11/pull/403#issuecomment-584224098) ? :D

@dnoliver I believe so, @pdxjohnny confirmed that this works.
